osfoffline/sync/remote.py

```diff
@@ -107,14 +113,27 @@ def _orphan_children(self, node, remote_children):
         locally for which the remote Node has been deleted are explicitly removed
         from OSFO's auditing and will be ignored.
         """
-        children_ids = map(lambda c: c.id, remote_children)
+        children_ids = [c.id for c in remote_children]
         for record in node.children:
             if record.id not in children_ids:
                 Session().delete(record)
+                logger.info("Deleted remotely deleted database Node<{}>".format(record.id))
```

Comment:
`record` is undefined inside this function; this logging message fails with an exception when run, and locks the progrsm. (as a shortcut, running `oinv flake` will catch this error)

```
[INFO][2016-01-21 10:03:39,062][Thread-3][osfoffline.sync.remote]: Beginning remote sync
[INFO][2016-01-21 10:03:39,450][Thread-3][requests.packages.urllib3.connectionpool]: Resetting dropped connection: test-api.osf.io
Exception in thread Thread-3:
Traceback (most recent call last):
  File "/Users/andyboughton/code/cos/OSF-Offline/osfoffline/sync/remote.py", line 128, in _preprocess_node
    remote_node = OSFClient().get_node(node.id)
  File "/Users/andyboughton/code/cos/OSF-Offline/osfoffline/client/osf.py", line 34, in get_node
    return Node.load(self.request_session, id)
  File "/Users/andyboughton/code/cos/OSF-Offline/osfoffline/client/osf.py", line 82, in load
    errors=data['errors']
osfoffline.client.osf.ClientLoadError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.5.0/Frameworks/Python.framework/Versions/3.5/lib/python3.5/threading.py", line 923, in _bootstrap_inner
    self.run()
  File "/Users/andyboughton/code/cos/OSF-Offline/osfoffline/sync/remote.py", line 76, in run
    self._preprocess_node(node, delete=False)
  File "/Users/andyboughton/code/cos/OSF-Offline/osfoffline/sync/remote.py", line 133, in _preprocess_node
    logger.info("Deleted remotely deleted database Node<{}>".format(record.id))
NameError: name 'record' is not defined


```